src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -0,0 +1,322 @@
+
+
+            // Do env variables if any
+            // We always want to set our defaults first and then set the profile ones. This permits a 
+            // profile to override ours
+
+            if (resolvedProfile.EnvironmentVariables != null && resolvedProfile.EnvironmentVariables.Count > 0)
+            {
+                foreach(var kvp in resolvedProfile.EnvironmentVariables)
+                {
+                    settings.Environment[kvp.Key] = kvp.Value;
+                }
+            }
+
+            settings.Executable = finalExecutable;
+            settings.Arguments = finalArguments;
+            settings.CurrentDirectory = workingDir;
+            settings.LaunchOperation = DebugLaunchOperation.CreateProcess;
+            settings.LaunchDebugEngineGuid = await GetDebuggingEngineAsync().ConfigureAwait(false); //.GetDebugEngineForFramework(runningFramework);
+            settings.LaunchOptions = launchOptions | DebugLaunchOptions.StopDebuggingOnEnd;
+            if(settings.Environment.Count > 0)
+            {
+                settings.LaunchOptions = settings.LaunchOptions | DebugLaunchOptions.MergeEnvironment;
+            }
+            return settings;
+        }
+
+        private async Task<Tuple<string, string>> GetRunnableProjectInformationAsync()
+        {
+            var properties = ConfiguredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+
+            // Assumes dotnet.exe is on the the path. This will be replaced by accessing a property from the project file
+            string executable, arguments;
+            var targetPath = await properties.GetEvaluatedPropertyValueAsync("TargetPath").ConfigureAwait(false);
+            if (targetPath.EndsWith(".exe"))
+            {
+                executable = targetPath;
+                arguments = "";
+            }
+            else
+            {
+                executable = GetFullPathOfExeFromEnvironmentPath("dotnet.exe");  // await properties.GetEvaluatedPropertyValueAsync("ExeToLaunch").ConfigureAwait(false);
+                if (executable == null)
+                {
+                    executable = "dotnet.exe";
+                }
+                arguments = targetPath.QuoteString();
+            }
+
+            return new Tuple<string, string>(executable, arguments);
+        }
+
+        private async Task<Guid> GetDebuggingEngineAsync()
+        {
+            var properties = ConfiguredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+            var framework = await properties.GetEvaluatedPropertyValueAsync("TargetFrameworkIdentifier").ConfigureAwait(false);
+
+            return ProjectDebuggerProvider.GetDebugEngineForFramework(framework);
+        }
+
+        /// <summary>
+        /// Searchs the path variable for the first match of exeToSearchFor. Returns 
+        /// null if not found. 
+        /// </summary>
+        public string GetFullPathOfExeFromEnvironmentPath(string exeToSearchFor)
```

Comment:
the messy part is the requirement for IFileSystem and IEnvironment. If could do an extension method of IFileSystem, or maybe IEnviroment but I'd still have to pass in the other interface. I'll move it as it could be useful in other places